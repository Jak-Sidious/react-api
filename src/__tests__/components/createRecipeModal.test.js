import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import { Card } from 'semantic-ui-react';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';

import ModalCreateRecipe from '../../components/commonComponents/createRecipeModal';

describe('ModalCreateRecipe component', () => {
  const del = jest.fn();
  const pathname = 'path';
  const props = {
    post: jest.fn(() => Promise.resolve('landing')),
    onChange: jest.fn(),
    handleChange: jest.fn(),
    history: { push: jest.fn() },
    location: { pathname },
    match: {
      param: {
        id: 1
      }
    }
  };
  const wrapper = shallow(<ModalCreateRecipe  {...props}/>);

  it('renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('creates recipe on click', () => {
    expect(wrapper.find('#createRec').length).toBe(1);
  })
});
