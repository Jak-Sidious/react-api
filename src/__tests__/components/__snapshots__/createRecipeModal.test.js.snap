// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCreateRecipe component renders properly without crashing 1`] = `
<Modal
  closeIcon={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
>
  <ModalHeader>
    Create Recipe
  </ModalHeader>
  <ModalContent>
    <Form
      as="form"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        label="Recipe Name"
        name="recipie_name"
        onChange={[Function]}
      />
      <FormTextArea
        as={[Function]}
        control={[Function]}
        label="Recipe ingredients"
        name="ingredients"
        onChange={[Function]}
      />
      <Button
        as="button"
        color="blue"
        id="createRec"
        onClick={[Function]}
      >
        Create
      </Button>
    </Form>
  </ModalContent>
</Modal>
`;
